src/test/java/fr/sncf/osrd/train/StaticSpeedLimitTest.java

```diff
@@ -129,8 +134,8 @@ public void simpleSpeedLimitTest() throws InvalidInfraException, SimulationError
                 .map(change -> (Train.TrainStateChange) change)
                 .collect(Collectors.toList());
         // Expect 2 state change: Train over starting Operational Point -> Go A to B -> Next phase
-        // The last 2 changes come from the end of phase point, and the last update when the phase ends
-        assertEquals(4, locationChanges.size());
+        // The change come from the train stop
+        assertEquals(3, locationChanges.size());
```

Comment:
I'd be a plus to know what 3 events are expected, either in a comment, or preferably, in an automated test

Reply:
There's only this comment left, and honestly I don't know what to do about it. It's not specific events, just the TrainStateChange. Their number depend on the internal interaction path.

I think the assert should be removed. It depends exclusively on internal implementation, there's no clear right number.